pkg/apis/triggers/v1alpha1/event_listener_validation_test.go

```diff
@@ -38,113 +37,276 @@ func Test_EventListenerValidate(t *testing.T) {
 		el   *v1alpha1.EventListener
 	}{{
 		name: "TriggerTemplate Does Not Exist",
-		el: bldr.EventListener("name", "namespace",
-			bldr.EventListenerSpec(
-				bldr.EventListenerTrigger("dne", "v1alpha1"))),
+		el: &v1alpha1.EventListener{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "name",
+				Namespace: "namespace",
+			},
+			Spec: v1alpha1.EventListenerSpec{
+				Triggers: []v1alpha1.EventListenerTrigger{{
+					Template: &v1alpha1.EventListenerTemplate{
+						Ref:        ptr.String("dne"),
+						APIVersion: "v1alpha1", // TODO: test with v1alpha1 version as well.
+					},
+				}},
+			},
+		},
 	}, {
 		name: "Valid EventListener No TriggerBinding",
-		el: bldr.EventListener("name", "namespace",
-			bldr.EventListenerSpec(
-				bldr.EventListenerTrigger("tt", "v1alpha1"))),
+		el: &v1alpha1.EventListener{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "name",
+				Namespace: "namespace",
+			},
+			Spec: v1alpha1.EventListenerSpec{
+				Triggers: []v1alpha1.EventListenerTrigger{{
+					Template: &v1alpha1.EventListenerTemplate{
+						Ref:        ptr.String("tt"),
+						APIVersion: "v1alpha1", // TODO: test with v1alpha1 version as well.
+					},
+				}},
+			},
+		},
 	}, {
 		name: "Valid EventListener with TriggerRef",
-		el: bldr.EventListener("name", "namespace",
-			bldr.EventListenerSpec(
-				bldr.EventListenerTriggerRef("tt"))),
+		el: &v1alpha1.EventListener{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "name",
+				Namespace: "namespace",
+			},
+			Spec: v1alpha1.EventListenerSpec{
+				Triggers: []v1alpha1.EventListenerTrigger{{
+					TriggerRef: "tt",
+				}},
+			},
+		},
 	}, {
 		name: "Valid EventListener with TriggerBinding",
-		el: bldr.EventListener("name", "namespace",
-			bldr.EventListenerSpec(
-				bldr.EventListenerTrigger("tt", "v1alpha1",
-					bldr.EventListenerTriggerBinding("tb", "TriggerBinding", "v1alpha1"),
-				))),
+		el: &v1alpha1.EventListener{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "name",
+				Namespace: "namespace",
+			},
+			Spec: v1alpha1.EventListenerSpec{
+				Triggers: []v1alpha1.EventListenerTrigger{{
+					Bindings: []*v1alpha1.EventListenerBinding{{
+						Ref:        "tb",
+						Kind:       v1alpha1.NamespacedTriggerBindingKind,
+						APIVersion: "v1alpha1", // TODO: APIVersions seem wrong?
```

Comment:
TODO is vague - I'm not sure what would need to be done to resolve this. Can you link to an issue / rephrase this to give more detail?